lib/renderer/window-setup.js

```diff
@@ -26,7 +26,7 @@
 const { defineProperty, defineProperties } = Object
 
 // Helper function to resolve relative url.
-const a = window.top.document.createElement('a')
+const a = window.document.createElement('a')
 const resolveURL = function (url) {
   a.href = url
   return a.href
```

Comment:
this function can be replaced with the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL)

```js
const resolveURL = function(url) {
  return new URL(url, location.href).href
}
```

Reply:
I'll follow up with this and other clean up referenced above.  I want to keep this down to just the new code